lib/game_team.rb

```diff
@@ -89,9 +89,52 @@ def self.worst_coach(season)
       coach_win_percentage.min_by { |coach, percent| percent }.first
     end
 
-    #best_offense
+    def self.total_goals_per_team
+      goals_per_team = Hash.new(0)
+      all_game_teams.each do |game_team|
+        goals_per_team[game_team.team_id] += game_team.goals
+      end
+      goals_per_team
+    end
+
+    def self.total_games_per_team
+      games_per_team = Hash.new(0)
+      all_game_teams.find_all do |game_team|
+        games_per_team[game_team.team_id] += 1
+      end
+      games_per_team
+    end
 
-    #worst offense
+    def self.average_goals_per_game
+      average_gpg = Hash.new(0)
+      all_game_teams.find_all do |game_team|
+        average_gpg[game_team.team_id] = (GameTeam.total_goals_per_team[game_team.team_id].to_f / GameTeam.total_games_per_team[game_team.team_id]).round(2)
+      end
+      average_gpg.delete(0)
```

Comment:
Is this how you got rid of the 0=>0?

Reply:
Yep! Probably a million other ways to do it but I simply removed that specific key/value pair.